### PR TITLE
VCBA-42: Added WebView to show privacy policy

### PR DESCRIPTION
**Pull request by amit-metri**
WebView was added to display a privacy policy.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -72,6 +72,8 @@ dependencies {
 
     implementation "androidx.constraintlayout:constraintlayout:$versions.constraint_layout"
 
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (pinakinkansara) [inline app/build.gradle:75]: We don't require this dependency, please remove it.

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/onboarding/OnBoardingFragment.kt [hunk 2]
```diff
@@ -53,9 +53,10 @@ class OnBoardingFragment : Fragment(R.layout.fragment_onboarding) {
             })
 
         binding.txtPrivacyPolicy.setSpannableLink {
-            val uri = Uri.parse("https://www2.gov.bc.ca/gov/content/covid-19/vaccine/proof/businesses#app-privacy-policy")
-            val intent = Intent(Intent.ACTION_VIEW, uri)
-            requireContext().startActivity(intent)
+            val uri = Uri.parse(getString(R.string.url_privacy_policy))
+            val action = OnBoardingFragmentDirections
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/vaxcheck/ui/onboarding/OnBoardingFragment.kt:57]: No need to pass URL from this screen as on that screen we can get it from strings.xml

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/scanner/BarcodeScannerFragment.kt [hunk 3]
```diff
@@ -127,9 +128,10 @@ class BarcodeScannerFragment : Fragment(R.layout.fragment_barcode_scanner), Scan
         }
 
         binding.txtPrivacyPolicy.setSpannableLink {
-            val uri = Uri.parse("https://www2.gov.bc.ca/gov/content/covid-19/vaccine/proof/businesses#app-privacy-policy")
-            val intent = Intent(Intent.ACTION_VIEW, uri)
-            requireContext().startActivity(intent)
+            val uri = Uri.parse(getString(R.string.url_privacy_policy))
+            val action = BarcodeScannerFragmentDirections
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/vaxcheck/ui/scanner/BarcodeScannerFragment.kt:132]: No need to pass from this screen, as we are going to get it from the required screen.

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/webview/WebViewFragment.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import ca.bc.gov.vaxcheck.utils.Helper
+import ca.bc.gov.vaxcheck.utils.viewBindings
+
+
+/**
+ * [WebViewFragment]
+ *
+ * @author amit metri
+ */
+class WebViewFragment : Fragment(R.layout.fragment_web_view) {
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/vaxcheck/ui/webview/WebViewFragment.kt:23]: Rename to privacy policy fragment.

File: app/src/main/res/navigation/bc_vax_nav_graph.xml [hunk 5]
```diff
@@ -15,17 +15,44 @@
...
+            app:popEnterAnim="@anim/nav_default_pop_enter_anim"
+            app:popExitAnim="@anim/nav_default_pop_exit_anim"/>
+    </fragment>
+
+    <fragment
+        android:id="@+id/webViewFragment"
+        android:name="ca.bc.gov.vaxcheck.ui.webview.WebViewFragment"
+        android:label="fragment_web_view"
+        tools:layout="@layout/fragment_web_view" >
+        <argument
```
Comment (pinakinkansara) [inline app/src/main/res/navigation/bc_vax_nav_graph.xml:52]: remove this argument

File: app/build.gradle (hunk 1 above)
Comment (amit-metri) [reply app/build.gradle:75]: added androidx lib for swiperefresh

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/onboarding/OnBoardingFragment.kt (hunk 2 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/vaxcheck/ui/onboarding/OnBoardingFragment.kt:57]: changes done.

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/scanner/BarcodeScannerFragment.kt (hunk 3 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/vaxcheck/ui/scanner/BarcodeScannerFragment.kt:132]: changes done.

File: app/src/main/java/ca/bc/gov/vaxcheck/ui/webview/WebViewFragment.kt (hunk 4 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/vaxcheck/ui/webview/WebViewFragment.kt:23]: changes done.

File: app/src/main/res/navigation/bc_vax_nav_graph.xml (hunk 5 above)
Comment (amit-metri) [reply app/src/main/res/navigation/bc_vax_nav_graph.xml:52]: removed.

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**